### DISCUSSION TITLE
Unable to select active EKS cluster during enrollment despite API authentication mode enabled

### DISCUSSION THREAD

**Discussion by spiffaz** [gravitational/teleport#49418] [Q&A] [answered]
When attempting to enroll an EKS cluster through the Teleport Cloud portal, the cluster appears in the list as active but remains greyed out and unselectable. The cluster has API authentication mode enabled and all required IAM permissions are configured.
Relevant details:

<img width="1055" alt="Screenshot 2024-11-25 at 18 19 30" src="https://github.com/user-attachments/assets/c7b61f10-8ab9-4cdf-bff7-a4d2532ad711">

Authentication modes: API, ConfigMap 
<img width="327" alt="Screenshot 2024-11-25 at 18 17 16" src="https://github.com/user-attachments/assets/811bc40e-314b-4b0d-9d71-1f2a72801d2c">

IAM policy attached to role: 
`
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "eks:ListClusters",
                "eks:DescribeCluster",
                "eks:ListAccessEntries",
                "eks:CreateAccessEntry",
                "eks:DeleteAccessEntry",
                "eks:AssociateAccessPolicy",
                "eks:TagResource"
            ],
            "Resource": "*"
        }
    ]
}
`

Teleport version is Teleport cloud

What could be missing and how can I fix this?

**Comment by webvictim** [comment]
@GavinFrazar might be able to make a suggestion here.

**Reply by GavinFrazar** [reply, marked as answer]
You have to hover over the EKS cluster to see a tooltip explaining why it's greyed out (we should make that tooltip more obvious).

That said, I'm guessing you need to enable public endpoint access.
Go to AWS console > EKS > your cluster > Networking, then:

![image](https://github.com/user-attachments/assets/617028ac-84f3-4570-88d4-65194d6817d1)

1. Click "manage endpoint access"
1. select the "public and private" option
1. expand the "advanced settings"
1. click "add source"
1. add "0.0.0.0/0" as a source CIDR block
1. click "save changes"

This allows public API endpoint access from 0.0.0.0/0 (i.e anywhere, but still requires auth of course) so that your Teleport Cloud instance can reach it and install the Teleport kube agent (Teleport Cloud does not publish our IP ranges, that's why we say use 0.0.0.0/0, sorry)

Once the Teleport kube agent is installed you don't need the public endpoint access anymore, so you can modify your cluster to only allow private API endpoint access again.

If you don't want to open up public endpoint access, then you have to install the teleport kube agent yourself with helm.
There's a guided enrollment wizard for that too, but it's called "Self-Hosted".
![image](https://github.com/user-attachments/assets/2eb5b5e5-0b4e-4b07-800b-b2539908df3c)

We also have a docs guide for doing it:
https://goteleport.com/docs/enroll-resources/kubernetes-access/register-clusters/iam-joining/
